frontend/common/src/helpers/router.tsx

```diff
@@ -87,35 +98,84 @@ export const pushToStateThenNavigate = (url: string, state: State): void => {
 };
 
 export interface RouterResult {
-  component: ReactElement;
+  component?: ReactElement;
   redirect?: string;
+  authorizedRoles?: Array<Role>;
 }
 
 export const useRouter = (
   routes: Routes<RouterResult>,
   missingRouteComponent: ReactElement,
+  notAuthorizedComponent: ReactElement,
+  isLoggedIn: boolean,
+  loggedInUserRoles: PossibleUserRoles,
 ): React.ReactElement | null => {
   const location = useLocation();
   const router = useMemo(() => new UniversalRouter(routes), [routes]);
   const [component, setComponent] = useState<React.ReactElement | null>(null);
   const pathName = location.pathname;
+  const apiRoutesRef = useRef(useApiRoutes()); // I'm not not sure why this function is unstable on its own!
+  const locale = getLocale(useIntl());
   // Render the result of routing
   useEffect((): void => {
     router
       .resolve(pathName)
-      .then(async (r) => {
-        // r may or may not be a promise, so attempt to resolve it. A non-promise value will simply resolve to itself.
-        const result = await Promise.resolve(r);
-        if (result?.redirect) {
-          redirect(result.redirect);
-        } else if (result) {
-          setComponent(result.component);
+      .then(async (routeMaybePromise) => {
+        // may or may not be a promise, so attempt to resolve it. A non-promise value will simply resolve to itself.
+        const route = await Promise.resolve(routeMaybePromise);
+
+        // handling a redirect
+        if (route?.redirect) {
+          redirect(route.redirect);
+        }
+
+        // is authorization required for this route?
+        const authorizationRequired = route?.authorizedRoles;
```

Comment:
Right now, the logic says that if they use needs to be logged in simply use an empty array.  A null means login not needed.  It's a bit "magicky" but I didn't really want to add another prop for this.  It's a good argument for adding a role like `USER` that everyone gets to test for being a logged in user.

Reply:
Yeah, I agree. Adding some basic role like `USER` would reduce confusion on `null` vs `[]`.

Reply:
Alternatively, remove this logic and use it as a pain point.  Anytime you have a page that needs a logged in user, use that as the opportunity to declare a new role.  For example, we need a role for CANDIDATE/APPLICANT/JOBSEEKER for the direct intake and profiles pages.

Reply:
I don't like the idea of a generic `USER` role, as it would be entirely redundant. But I'm happy with adding a `CANDIDATE/APPLICANT` role, and try to keep it so that any page requiring login requires a role.

Reply:
Sounds good.  #2911 created.